src/main/java/de/tum/in/www1/artemis/web/rest/hestia/ProgrammingExerciseTaskResource.java

```diff
@@ -55,7 +55,7 @@ public ResponseEntity<Set<ProgrammingExerciseTask>> getTasks(@PathVariable Long
         ProgrammingExercise exercise = programmingExerciseRepository.findByIdElseThrow(exerciseId);
         authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.TEACHING_ASSISTANT, exercise, null);
 
-        Set<ProgrammingExerciseTask> tasks = programmingExerciseTaskRepository.findByExerciseIdWithTestCaseAndSolutionEntriesElseThrow(exerciseId);
+        Set<ProgrammingExerciseTask> tasks = programmingExerciseTaskService.getTasksWithUnassignedTestCases(exerciseId);
```

Comment:
This change does not fare well with exercise hints currently, as this endpoint is also used by the client for exercise hints.
The issue with this is that when you try to create or update a hint and set the assigned task to `Not assigned to any task` there will be an internal server error, as this task doesn't actually exist, as it is made up in the service method.
<img width="1112" alt="grafik" src="https://user-images.githubusercontent.com/18218285/227723040-f7c5cdf4-29ec-4dcf-b741-871117bbf234.png">

You will need to handle this somehow and make sure that this behavior is also documented in the client, as it isn't what you would normally expect.
Personally, I even think creating a separate endpoint for this might be good, just so that this endpoint still does what it is supposed to.
An alternative would also be to create a new `GET test-cases` endpoint that also eagerly fetches the tasks of the test cases. This way, you would need one less server call, as you currently still call `GET test-cases` in addition to `GET tasks`.

Reply:
- UI change in https://github.com/ls1intum/Artemis/pull/6344/commits/64a6662d69d151489dc25ab530968245be29b441
- Endpoint change in https://github.com/ls1intum/Artemis/pull/6344/commits/17a6651c41ff388343af18c9226c6075247f3d2c